### PR TITLE
azure_rm_virtualmachine: add custom image support

### LINKED ISSUE #24328: Not able to create new instance with custom user image in azure with Ansible playbook.
##### ISSUE TYPE

 - Bug Report
 - Feature Idea

##### COMPONENT NAME
azure_rm_virtualmachine

##### ANSIBLE VERSION

```
ansible 2.3.0.0
```

##### SUMMARY
I am not able to create instance from my custom image in azure but at the same time i am able to create instance with available images in the market.

##### STEPS TO REPRODUCE

```yaml
    - name: Create VM in azure
      azure_rm_virtualmachine:
        client_id: <my client id>
        tenant: <My tenant id>
        secret: <my secret key>
        subscription_id: <my subscription>
        resource_group: my-resource-group
        image: <my custom image name>
        name: ansibleTestVM
        location: 'eastus2'
        admin_username: 'myusername'
        admin_password: 'mypassword'
        os_type: Linux
        storage_account_name: mystorage
        vm_size: Standard_D1
        tags:
          Name: Ansible_test_tag
          Queqe: Q4
        network_interface_names: 'ansibleTestVM1-nic'
```

##### ACTUAL RESULTS

```

```

### PR DESCRIPTION
**Pull request by jborean93**
##### SUMMARY
Allows someone to create a VM from a custom image using the new `custom_image` option.

Fixes https://github.com/ansible/ansible/issues/24328

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
azure_rm_virtualmachine

##### ANSIBLE VERSION
```
2.5
```

### CODE REVIEW

**Comment by jborean93** [comment]
@mattclay I would love to create some integration tests for this but unsure really how to go about doing it. I know the commands to run through the azure-cli library to create an image from an existing VM but there is no module equivalent right now. Any suggestions would be great and I am happy to try it

**Comment by mattclay** [comment]
@jborean93 I'd check to see if anyone working on Azure modules has plans to implement what is needed for the tests. If so, it would be reasonable to wait for that. If not, we could either create a minimal module private to the tests to get the job done, or have the tests install the Azure CLI tools and use those.

**Comment by jborean93** [comment]
@mattclay thanks for the advice, after creating this PR I've seen a new PR being added that actually does what I'm looking for (quite fortunate). When it is merged I'll add some integration tests.

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py [hunk 1]
```diff
@@ -92,9 +92,19 @@
...
+            - If a string, the image is sourced from a custom image based on the
+              name.
+            - 'If a dict with the keys C(publisher), C(offer), C(sku), and
+              C(version), the image is sourced from a Marketplace image. NOTE:
+              set image.version to C(latest) to get the most recent version of a
+              given image.'
+            - 'If a dict with the keys C(name) and C(resource_group), the image
+              is sourced from a custom image based on the C(name) and
+              C(resource_group) set. NOTE: the key C(resource_group) is optional
+              and if omitted, all images in the subscription will be search for
```
Comment (nitzmahone) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py:105]: s/search/searched/

**Review by nitzmahone — APPROVED** [review]

**Comment by nitzmahone** [comment]
rebuild_merge

**Comment by radixdlt** [comment]
Thanks for this. I created follow up PR #33422 for azure_rm_virtualmachine_scaleset .

**Pull request merged**